### ISSUE TITLE
pqmf synthesis filter

### ISSUE THREAD

**Issue by superhg2012** [TensorSpeech/TensorflowTTS#68]
Hi,  why  synthesis filter still use h_analysis here?   
```
  # [subbands, 1, taps + 1] == [filter_width, in_channels, out_channels]
        analysis_filter = np.expand_dims(h_analysis, 1)
        analysis_filter = np.transpose(analysis_filter, (2, 1, 0))
        synthesis_filter = np.expand_dims(h_analysis, 0)
        synthesis_filter = np.transpose(synthesis_filter, (2, 1, 0))
```

**Comment by dathudeptrai** [comment]
@superhg2012 it's different at exand_dims op, you can see

```
analysis_filter = np.expand_dims(h_analysis, 1)
```
while
```
 synthesis_filter = np.expand_dims(h_analysis, 0)
```

**Comment by superhg2012** [comment]
@dathudeptrai  i mean whether  h_analysis should be  h_synthesis ?

**Comment by dathudeptrai** [comment]
@superhg2012 no, it's still h_analysis. Synthesis is the inverse step of analysis, so we just need change the order of the filter.

**Comment by superhg2012** [comment]
there is some diffenence with here [](https://github.com/kan-bayashi/ParallelWaveGAN/blob/master/parallel_wavegan/layers/pqmf.py)

**Comment by dathudeptrai** [comment]
@superhg2012 it's exactly the same. Pytorch/Tensorflow filter weight have different order :)). In tensor it's [filter_width, in_channels, out_channels] while pytorch is [out_channels, in_channels, filter_width] if i have no wrong :)).

**Comment by superhg2012** [comment]
> @superhg2012 it's different at exand_dims op, you can see
> 
> ```
> analysis_filter = np.expand_dims(h_analysis, 1)
> ```
> 
> while
> 
> ```
>  synthesis_filter = np.expand_dims(h_analysis, 0)
> ```

here 

 `synthesis_filter = np.expand_dims(h_analysis, 0)`

synthesis_filter  is used to sum sub_band signal into full band signal , then my point is  why  synthesis_filter  use  h_analysis  ?   why not h_synthesis ?

**Comment by superhg2012** [comment]
get it!! thanks!

**Comment by superhg2012** [comment]
Another problem, my multi_band melgan generated audio like below. I am training with 16000 sampling rate.  there is line around 4K sampleing rate.
what will cause this ?
![image](https://user-images.githubusercontent.com/8817309/85998116-f9cd3200-ba3c-11ea-9a4a-26b0ac1f6b57.png)

**Comment by dathudeptrai** [comment]
@superhg2012 longer training will solve it :))) don’t worry :3

**Comment by superhg2012** [comment]
all right, expect it !!

**Comment by superhg2012** [comment]
after running nearly 500K steps, disconnections still exist. worrying...

![image](https://user-images.githubusercontent.com/8817309/86308591-4dfa3100-bc4c-11ea-93dd-3a25b8ac10e7.png)

**Comment by dathudeptrai** [comment]
@superhg2012 don't worry :))) 400k steps is just apply for LJspeech :)). Maybe ur dataset is larger :)) can you share me a tensorboard :d

**Comment by superhg2012** [comment]
I am running a 6.4 hours Chinese dataset. far smaller than ljspeech dataset .  I am using  starting learning rate 1e-3, so loss decreased fast at very begining, after 40K steps, it grows very large. So I gess this result in the loss explode.

![image](https://user-images.githubusercontent.com/8817309/86309121-8cdcb680-bc4d-11ea-8655-d8f461083181.png)

**Comment by dathudeptrai** [comment]
@superhg2012 you are not using our code ? right ?

**Comment by Maxxiey** [comment]
got same problem here, trained on my own dataset of 1000 sentences(0.76 hour) for 500k.

mels and tensorboard chats:
![image](https://user-images.githubusercontent.com/42697295/86309128-8f3f1080-bc4d-11ea-9862-653eff347448.png)
![image](https://user-images.githubusercontent.com/42697295/86309248-d5946f80-bc4d-11ea-9451-6183b1af9d1d.png)

**Comment by dathudeptrai** [comment]
@Maxxiey ur dataset too small, look at validation loss, you are overfiting now :))

**Comment by superhg2012** [comment]
> @superhg2012 you are not using our code ? right ?

yes, I am using my own code, but I have compared with this repo. almost the same.  Just talk  about this mb-melgan issue!

**Comment by dathudeptrai** [comment]
@superhg2012 hmm, i don't know if the amount of data affect this :)) maybe training longer can solve or maybe not :D. BTW, we are training Baker chinese dataset and will support end2end chinese soon :D. After that maybe you can use our pretrained chinese model to train ur model.

**Comment by superhg2012** [comment]
great, by the way , will 16000 sampling rate configs supported?

**Comment by dathudeptrai** [comment]
@superhg2012 yes, i use this framework for my private dataset 16k, you just need change the sampling_rate on preprocessing.

**Comment by Maxxiey** [comment]
@dathudeptrai thanks a lot, looking forward to the pretrained model :D, and yes I will let the training continue and check the result later.

**Comment by superhg2012** [comment]
get it!   Another question , how  will n_fft  affect the model ?  like n_fft = 2048 and n_fft=1024 ?

**Comment by dathudeptrai** [comment]
@superhg2012  i haven't tune those parameter yet :)) just follow almost preprocessing params on github repo for ljspeech :v. I'm not goot at signal processing, my major is computer vision and NLP, so ...

**Comment by superhg2012** [comment]
@dathudeptrai  whatever, this repo is a great work！

**Comment by superhg2012** [comment]
after pretraining G 200K steps and correct training procedures errors, the disconnections is disappearing now after 300K steps for both nfft=2048 and nfft=1024 16000 sr.
![image](https://user-images.githubusercontent.com/8817309/86502737-0228c280-bdd9-11ea-8750-a8b38f95ff2d.png)

**Comment by dathudeptrai** [comment]
What is ur error and how you fix it?

**Comment by superhg2012** [comment]
after training G 200K steps,  G and D are training at same but not  alternately！
